src/lib/Uploader.ts

```diff
@@ -46,7 +47,11 @@ export default class Uploader {
     };
 
     if (this.options.config) {
-      AWS.config.loadFromPath(this.options.config);
+      if (typeof this.options.config === 'string') {
+        AWS.config.loadFromPath(this.options.config);
+      } else if (typeof this.options.config === 'object') {
+        AWS.config.constructor(this.options.config);
+      }
```

Comment:
Can we add an `else` with an error that an unsupported Config object is passed, so not used?

Reply:
done